File: src/main/java/kitchenpos/eatinorders/tobe/entity/OrderLineItems.java
package kitchenpos.eatinorders.tobe.entity;

import jakarta.persistence.*;

import java.util.List;

@Embeddable
public class OrderLineItems {
    @OneToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinColumn(
            name = "order_id",
            nullable = false,
            columnDefinition = "binary(16)",
            foreignKey = @ForeignKey(name = "fk_order_line_item_to_orders")
    )
    private List<OrderLineItem> orderLineItems;

    protected OrderLineItems() {}

    public OrderLineItems(List<OrderLineItem> orderLineItems) {
        this.orderLineItems = orderLineItems;
    }
}

File: src/main/java/kitchenpos/eatinorders/tobe/OrderLineItems.kt
package kitchenpos.eatinorders.tobe

class OrderLineItems(
    val orderLineItemList: List<OrderLineItem>,
) {
    init {
        require(orderLineItemList.isNotEmpty()) { "주문 상품은 1개 이상이어야 합니다." }
    }
}
